### ISSUE TITLE
Meta.ddl() exports broken DDL for materialized views

### ISSUE THREAD

**Issue by mgramin** [jOOQ/jOOQ#15531]
### Expected behavior

We expect that Meta.ddl() exports a valid DDL for materialized views:
```sql
create materialized view test_mat_view as
select * from test_table;
```

### Actual behavior

Meta.ddl() exports broken DDL for that views:
```sql
create view "public"."test_mat_view"("id")
as
```

### Steps to reproduce the problem

Create a simple table and related materialized view in DB:
```sql
create table test_table (
  id int not null
);

create materialized view test_mat_view as
select * from test_table;
```

Run this code:
```java
        Connection conn = DriverManager.getConnection("jdbc:postgresql://localhost:6000/postgres", "postgres", "postgres");
        Configuration configuration = new DefaultConfiguration().set(conn).set(SQLDialect.POSTGRES);
        Meta meta = using(configuration).meta();

        Arrays.stream(meta.filterSchemas(v -> v.getName().equalsIgnoreCase("public"))
                        .ddl()
                        .queries())
                .map(Object::toString)
                .forEach(System.out::println);
```

jOOQ will generate a broken DDL for the materialized view:
```sql
create view "public"."test_mat_view"("id")
as
create table "public"."test_table" (
  "id" int4 not null
)
```

### jOOQ Version

jOOQ Professional Edition 3.18.6

### Database product and version

PostgreSQL 15.2 (Debian 15.2-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit

### Java Version

openjdk 17.0.2 2022-01-18

### OS Version

Microsoft Windows [Version 10.0.19044.2846]

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.6.0

**Comment by mgramin** [comment]
Thank you

**Comment by lukaseder** [comment]
Just an update that I'm shaving this yak here: https://github.com/jOOQ/jOOQ/issues/9483. It'll still take some polishing, but I'm positive that I can close this issue next week.

**Comment by lukaseder** [comment]
This has already been implemented in jOOQ 3.19.0 (via other issues, too). Not sure why I hadn't closed this one at the time.
